### PR TITLE
YahooSSP: Added video support.

### PR DESCRIPTION
**Pull request by oath-jac**

### CODE REVIEW

File: adapters/yahoossp/yahoossp.go [hunk 1]
```diff
@@ -156,13 +159,20 @@ func changeRequestForBidService(request *openrtb2.BidRequest, extension *openrtb
...
 
-	banner := *request.Imp[0].Banner
-	request.Imp[0].Banner = &banner
+	return nil
+}
 
+func validateBanner(banner *openrtb2.Banner) error {
 	if banner.W != nil && banner.H != nil {
 		if *banner.W == 0 || *banner.H == 0 {
 			return fmt.Errorf("Invalid sizes provided for Banner %dx%d", *banner.W, *banner.H)
```
Comment (VeronikaSolovei9) [inline adapters/yahoossp/yahoossp.go:178]: Can you please add a test coverage for this case?

Comment (VeronikaSolovei9) [inline adapters/yahoossp/yahoossp.go:178]: Can you please add a test coverage for this case?

Comment (oath-jac) [reply adapters/yahoossp/yahoossp.go:178]: I added new tests that covers the rest of the validateBanner execution paths.

**Review by VeronikaSolovei9 — APPROVED** [review]
Thank you for addressing a comment. LGTM

File: adapters/yahoossp/yahoossp.go [hunk 2]
```diff
@@ -156,13 +159,20 @@ func changeRequestForBidService(request *openrtb2.BidRequest, extension *openrtb
...
-	if request.Imp[0].Banner == nil {
-		return nil
+	if request.Imp[0].Banner != nil {
+		banner := *request.Imp[0].Banner
+		request.Imp[0].Banner = &banner
+
+		err := validateBanner(&banner)
+		if err != nil {
+			return err
+		}
```
Comment (AlexBVolcy) [inline adapters/yahoossp/yahoossp.go:169]: I'm interested in why this code was updated for this PR? Does it help support video in some way, or did you just want to tweak how Banner is supported?

**Review by AlexBVolcy — COMMENTED** [review]
Looks good. Just left one comment.

File: adapters/yahoossp/yahoossp.go (hunk 2 above)
Comment (oath-jac) [reply adapters/yahoossp/yahoossp.go:169]: Just some small internal refactoring, to move banner validation into a particular function. I think it improves the readability.

**Review by AlexBVolcy — APPROVED** [review]

**Comment by bsardo** [comment]
Hi @oath-jac, it looks like you might need to make a change to the Yahoo SSP section of the [prebid docs](https://docs.prebid.org/dev-docs/pbs-bidders.html#yahoossp). I see that it lists both banner and video as supported media types in the "Features" subsection but there is another subsection "Important Notice (JS vs PBS)" that states that PBS does not support video inventory, which I think needs to be removed now.
Can you please open a docs PR here? https://github.com/prebid/prebid.github.io
Once you've opened that we can merge this PR. Thanks!

**Comment by oath-jac** [comment]
I opened this PR https://github.com/prebid/prebid.github.io/pull/3664

thank you.

**Pull request merged**